This is a Repository for the data used for the paper: 
# Genomic evidence for gene flow between monarchs with divergent migratory phenotypes and flight performance 
![alt text](https://github.com/venta380/Monarch_genomics/blob/master/logo.png "Logo Title Text 1")



The scripts used for mapping the population resequencing data to the reference genome and the SNP calling can be found in the directory “SNP_calling”

The final set of genotypes used in this study can be found in here:
[link to download the VCF file]( https://www.dropbox.com/s/sawz9bn1sfc0fii/recal_snps_PASS_only_removed_repeats_20190315.vcf.gz?dl=0)

The reference genome assembly used in this project is a version 3 scaffolds from Monarchbase:
[link to download the reference genome assembly file]( http://monarchbase.umassmed.edu/download/Dp_genome_v3.fasta.gz)

Genomic positions extracted from the genome annotation can be downloaded from: 
[link to download intron genomic positions]( https://www.dropbox.com/s/g2zlr4coptp42s1/introns.csv.gz?dl=0)
[link to download Codon position 1 genomic positions]( https://www.dropbox.com/s/l9ct1vs32bu9dyb/codon_df_1.csv.gz?dl=0)
[link to download Codon position 2 genomic positions]( https://www.dropbox.com/s/nn6109bjfdm3xjw/codon_df_2.csv.gz?dl=0)
[link to download Codon position 3 genomic positions]( https://www.dropbox.com/s/fb22tvf5v1lg4ca/codon_df_3.csv.gz?dl=0)

The file dadi_temp.py has the script to run demographic history using dadi. 

Monarch_east_west_plots.py script give the genome scan plots 
