# Here we provide the scripts to calculate the nucleotide diversity of  Eastern and Western monarchs for different genomic positions. 

Genomic positions were classified as Intergenic, Intronic, 1st, 2nd, 3rd codon positions and 4-fold degenerate sites (4D).

# Monarch_Pi_all_sites.py
The script Monarch_Pi_all_sites.py generates the data base of nucleotide diversity seperatly for Intergenic, Intronic, 1st, 2nd, 3rd codon positions and 4-fold degenerate sites (4D). These results are provided in Pi_data.csv

# Monarch_plots_pi.py
This script will give the folowing plot
![alt text](https://github.com/venta380/Monarch_genomics/blob/master/Nucleotide%20diversity/Screenshot%202020-01-27%2012.34.54.png "Logo Title Text 1")

# Monarch_pi_tw_td_sfs_all_sites_1977.py
This script will calculate the nucleotide diversity (π), Watterson’s θ and Tajima’s D for the specific sites (All, intergenic, codon positions, 0fold and 4fold) in the genome. 

